library/oneview_volume.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/python
+
+###
+# Copyright (2016) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+from ansible.module_utils.basic import *
+from hpOneView.oneview_client import OneViewClient
+
+DOCUMENTATION = '''
+---
+module: oneview_volume
+short_description: Manage OneView Volume resources.
+description:
+    - Provides an interface to manage Volume resources. It allows create, update, delete or repair the volume, and
+      create or delete a snapshot.
+requirements:
+    - "python >= 2.7.9"
+    - "hpOneView"
+author: "Mariana Kreisig (@marikrg)"
+options:
+    config:
+      description:
+        - Path to a .json configuration file containing the OneView client configuration.
+      required: true
+    state:
+        description:
+            - Indicates the desired state for the Volume resource.
+              'present' creates or adds the resource when it does not exist, otherwise it updates the resource. The
+              update operation is non-idempotent.
```

Comment:
It's not idempotent because the update always will be called when the resource already exists. There is no way to compare the existent resource and the given options.